File: javatest/src/main/java/com/anymore/javatest/di/component/subcomponent/MainActivitySubComponent.java
package com.anymore.javatest.di.component.subcomponent;

import com.anymore.javatest.MainActivity;
import com.anymore.javatest.di.module.MainActivityViewModelModule;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

/**
 * Created by liuyuanmao on 2019/2/22.
 */
@Subcomponent(modules = {MainActivityViewModelModule.class,MainActivityViewModelModule.class})
public interface MainActivitySubComponent extends AndroidInjector<MainActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MainActivity>{}
}

File: app/src/main/java/com/anymore/simple/di/component/subcomponent/MainActivitySubComponent.kt
package com.anymore.simple.di.component.subcomponent

import com.anymore.simple.mvvm.view.MainActivity
import com.anymore.simple.di.module.MainViewModelModule
import dagger.Subcomponent
import dagger.android.AndroidInjector

/**
 * Created by liuyuanmao on 2019/2/21.
 */
@Subcomponent(modules = [MainViewModelModule::class])
interface MainActivitySubComponent :AndroidInjector<MainActivity>{

    @Subcomponent.Builder
    abstract class Builder: AndroidInjector.Builder<MainActivity>()
}